src/main/java/com/example/auctionapp/entity/Category.java

```diff
@@ -27,12 +29,14 @@
     @NonNull
     private String name;
 
-    @ManyToOne(
+    @OneToMany(
```

Comment:
Isn't this ManyToOne, if it is otherwise that would mean one category can have many supercategories which is not true